src/docs/spec-model.md

```diff
@@ -81,3 +81,68 @@ In addition, predicates from the following sources MAY also be encouraged:
 
 * any relation from the [Relation Ontology (RO)](https://obofoundry.org/ontology/ro.html);
 * any relation under [skos:mappingRelation](http://www.w3.org/2004/02/skos/core#mappingRelation) in the [Semantic Mapping Vocabulary](https://mapping-commons.github.io/semantic-mapping-vocabulary/).
+
+## Non-standard slots
+
+<a id="non-standard-slots"></a>
+
+Implementations are only REQUIRED to support the standard metadata slots defined in the SSSOM LinkML schema.
+
+However, implementations MAY support the use of supplementary, non-standard slots (hereafter called _extension slots_ or simply _extensions_). There are two types of extension slots: _defined_ extension slots and _undefined_ extension slots.
+
+### Defined extensions
+
+Defined extensions are non-standard slots that are explicitly declared (or, _defined_) before being used. Implementations SHOULD support the use of defined extensions.
+
+Extensions are defined in the `extension_definition` slot of the `MappingSet` object. Each definition is comprised of three elements:
+
+* the name of the slot, as it will appear when used in a mapping set (`slot_name`);
+* a property intended to specify the meaning of the slot (`property`);
+* the type of values expected by the slot (`type_hint`).
+
+A definition MUST have at least a `slot_name`. The name MUST be a XML “non-colonized name” (“NCName”, see [Namespaces in XML, §2](https://www.w3.org/TR/1999/REC-xml-names-19990114/#NT-NCName)). The name MUST NOT match the name of an existing standard slot.
+
+To avoid any conflicy with a future version of the SSSOM specification (which could introduce new standard slot names), implementations are strongly encouraged to craft extension slot names that start with the `ext_` prefix. No new standard slot with a name starting with `ext_` will ever be introduced in any future version of the standard. (This is an advice for SSSOM producers only; SSSOM consumers MUST NOT reject an extension slot solely on the basis that its name does not start with `ext`.)
+
+A definition SHOULD have a `property`. If it does not, implementations MUST automatically construct a default property by catenating the prefix `http://sssom.invalid/` with the name of the extension.
+
+The slot name and the property MUST be unique to each definition. No two definitions can share the same name and/or the same property.
```

Comment:
> No two definitions can share the same name and/or the same property.

Why property? Typical example:

- nico_comment
- damien_comment
- harshad_comment

all have the rdfs:comment property?

Reply:
(As an aside, while I think this part of the definition should be dropped, there is a bit of a risk that the property slot in the extensions clashes with the "meaning" or "slot_uri" values in the regular sssom slots.

Reply:
> Why property? Typical example:
>
>    nico_comment
>    damien_comment
>    harshad_comment
>
> all have the rdfs:comment property?

No! The _property_, not the *slot_name*, is what identifies the extension value. That’s why it is strongly recommended that every extension definition has a _property_, and why implementations MUST construct one automatically if needed. The slot name is merely how the extension is referred to in the SSSOM/TSV format.

You’re the one who insisted on that because it was critical for you that extension values could be serialised into RDF, which required them to have a proper IRI instead of merely a name. How would you serialise your “typical example” in RDF if your three extension slots all have the same `rdfs:comment` property???

Reply:
Ok I dont remember all the details on this discussion, I apologise. I was happy with our final draft so I wont change my mind now!